arangod/Cluster/AgencyCallback.cpp

```diff
@@ -92,7 +93,7 @@ void AgencyCallback::refetchAndUpdate(bool needToAcquireMutex, bool forceCheck)
 void AgencyCallback::checkValue(std::shared_ptr<VPackBuilder> newData, bool forceCheck) {
   // Only called from refetchAndUpdate, we always have the mutex when
   // we get here!
-  if (!_lastData || !_lastData->slice().equals(newData->slice()) || forceCheck) {
+  if (!_lastData || arangodb::basics::VelocyPackHelper::compare(_lastData->slice(), newData->slice(), false) != 0 || forceCheck) {
```

Comment:
Maybe on should also provide `arangodb::basics::VelocyPackHelper::equals` which is implemented as `compare == 0`. Because someday someone will again forget the `== 0`.